packages/react/src/errorboundary.tsx

```diff
@@ -160,14 +160,18 @@ class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundarySta
     const { error, componentStack, eventId } = this.state;
 
     if (error) {
-      if (React.isValidElement(fallback)) {
-        return fallback;
-      }
+      let element: React.ReactElement | undefined = undefined;
       if (typeof fallback === 'function') {
-        return fallback({ error, componentStack, resetError: this.resetErrorBoundary, eventId }) as FallbackRender;
+        element = fallback({ error, componentStack, resetError: this.resetErrorBoundary, eventId });
+      } else {
+        element = fallback;
+      }
+
+      if (React.isValidElement(element)) {
+        return element;
       }
 
-      // Fail gracefully if no fallback provided
+      // Fail gracefully if no fallback provided or is not valid
```

Comment:
We can use `logger` from `@sentry/utils` to log out a warning here: https://github.com/getsentry/sentry-javascript/blob/master/packages/utils/src/logger.ts

```ts
import { logger } from `@sentry/utils`
```

Reply:
Perfect, thanks!